src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordObjectMethod.cs

```diff
@@ -16,25 +18,49 @@ protected SynthesizedRecordObjectMethod(SourceMemberContainerTypeSymbol containi
         {
         }
 
+        protected sealed override DeclarationModifiers MakeDeclarationModifiers(DeclarationModifiers allowedModifiers, DiagnosticBag diagnostics)
+        {
+            const DeclarationModifiers result = DeclarationModifiers.Public | DeclarationModifiers.Override;
+            Debug.Assert((result & ~allowedModifiers) == 0);
+            return result;
+        }
+
         protected sealed override void MethodChecks(DiagnosticBag diagnostics)
         {
             base.MethodChecks(diagnostics);
+            VerifyOerridesMethodFromObject(this, diagnostics);
+        }
 
-            var overridden = OverriddenMethod?.OriginalDefinition;
+        /// <summary>
+        /// Returns true if reported an error
+        /// </summary>
+        internal static bool VerifyOerridesMethodFromObject(MethodSymbol overriding, DiagnosticBag diagnostics)
+        {
+            bool reportAnError = false;
 
-            if (overridden is null || (overridden is SynthesizedRecordObjEquals && overridden.DeclaringCompilation == DeclaringCompilation))
+            if (!overriding.IsOverride)
             {
-                return;
+                reportAnError = true;
             }
+            else
+            {
+                var overridden = overriding.OverriddenMethod?.OriginalDefinition;
 
-            MethodSymbol leastOverridden = GetLeastOverriddenMethod(accessingTypeOpt: null);
+                if (overridden is object && !(overridden.ContainingType is SourceMemberContainerTypeSymbol { IsRecord: true } && overridden.ContainingModule == overriding.ContainingModule))
+                {
+                    MethodSymbol leastOverridden = overriding.GetLeastOverriddenMethod(accessingTypeOpt: null);
 
-            if (leastOverridden is object &&
-                leastOverridden.ReturnType.SpecialType == ReturnType.SpecialType &&
-                leastOverridden.ContainingType.SpecialType != SpecialType.System_Object)
+                    reportAnError = leastOverridden.ReturnType.SpecialType == overriding.ReturnType.SpecialType &&
```

Comment:
>leastOverridden.ReturnType.SpecialType == overriding.ReturnType.SpecialType [](start = 36, length = 75)

When can this be false?

Reply:
Or more specifically, when can this be false the next check be true?

---
In reply to: [447230644](https://github.com/dotnet/roslyn/pull/45518#discussion_r447230644) [](ancestors = 447230644)

Reply:
> Or more specifically, when can this be false but the next check be true?

When the object method is shadowed by a method in the base. BTW, this condition didn't change since the last PR that you reviewed, I think.

---
In reply to: [447237880](https://github.com/dotnet/roslyn/pull/45518#discussion_r447237880) [](ancestors = 447237880,447230644)

Reply:
> BTW, this condition didn't change since the last PR that you reviewed, I think.

Then I missed asking this in the previous PR.

> When the object method is shadowed by a method in the base.

If the method is shadowed in the base, then the leastOverridden here wouldn't be defined in `System.Object`, right? I'm not seeing why we can't just check if the method is overriding a method from `System.Object` and leave it at that.

---
In reply to: [447266797](https://github.com/dotnet/roslyn/pull/45518#discussion_r447266797) [](ancestors = 447266797,447237880,447230644)

Reply:
> If the method is shadowed in the base, then the leastOverridden here wouldn't be defined in System.Object, right? I'm not seeing why we can't just check if the method is overriding a method from System.Object and leave it at that.

Because if the return type doesn't match, we complain about that elsewhere with more precise error. This error would look like saying the same thing, but with different words. 


---
In reply to: [447305823](https://github.com/dotnet/roslyn/pull/45518#discussion_r447305823) [](ancestors = 447305823,447266797,447237880,447230644)

Reply:
Ahh, I see. I was misreading the condition, thanks.

---
In reply to: [447307840](https://github.com/dotnet/roslyn/pull/45518#discussion_r447307840) [](ancestors = 447307840,447305823,447266797,447237880,447230644)